src/wallet/connection.ts

```diff
@@ -88,4 +91,95 @@ class WalletServiceConnection extends BaseConnection {
   }
 }
 
-export default WalletServiceConnection;
+/**
+ * This connection is used when the wallet has websocket disabled.
+ */
+export class DummyWalletServiceConnection extends EventEmitter implements IConnection {
+  public readonly isDummyConnection = true;
+  public walletId?: string;
+
+  private currentNetwork: string;
+  private currentServer: string;
+  protected state: ConnectionState;
+
+  constructor(options: WalletServiceConnectionParams = { walletId: 'dummy-wallet' }) {
+    super();
+
+    const {
+      walletId,
+      network,
+      servers,
+    } = {
+      ...DEFAULT_PARAMS,
+      ...options,
+    };
+
+    if (!network) {
+      throw Error('You must explicitly provide the network.');
+    }
+
+    this.walletId = walletId;
+    this.currentNetwork = network || 'dummynet';
+    this.state = ConnectionState.CLOSED;
+    this.currentServer = servers[0] || config.getServerUrl();
+  }
+
+  start(): void {
+    if (!this.walletId) {
+      throw new Error('Wallet id should be set before connection start.');
+    }
+  }
+
+  stop(): void {
+    this.removeAllListeners();
```

Comment:
Why do you have to implement all the methods? I thought it would be a dummy connection but here you have change of state.

Reply:
It mimics a real implementation but with no logic underneath. Here I'm removing the listeners because the class extends the EventEmmiter. Eventually, if the wallet set any listener, it will be removed when `stop` is called. But right now the `start` is not even been called. Therefore this implementation is a guard *just in case*.